behaviour/graql/language/insert.feature

```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
+      """
+    When the integrity is validated
+    Then get answers of graql query
+      """
+      match $c has hex-value "#FF0000"; get;
+      """
+    Then concept identifiers are
+      |     | check | value      |
+      | COL | value | colour:red |
+    Then uniquely identify answer concepts
+      | c   |
+      | COL |
 
-  Scenario: if any insert in a transaction fails with a `key` violation, none of the inserts are performed
 
+  Scenario: when inserting an additional attribute value on an attribute, the type of the owning attribute can be optionally specified
+    Given graql define
+      """
+      define
+      colour sub attribute, value string, has hex-value;
+      hex-value sub attribute, value string;
+      """
+    Given the integrity is validated
+    Given graql insert
+      """
+      insert
+      $c "red" isa colour;
+      """
+    Given the integrity is validated
+    Given get answers of graql query
+      """
+      match $c has hex-value "#FF0000"; get;
+      """
+    Given answer size is: 0
+    When graql insert
+      """
+      match
+        $c "red" isa colour;
+      insert
+        $c isa colour, has hex-value "#FF0000";
+      """
+    When the integrity is validated
+    Then get answers of graql query
+      """
+      match $c has hex-value "#FF0000"; get;
+      """
+    Then concept identifiers are
+      |     | check | value      |
+      | COL | value | colour:red |
+    Then uniquely identify answer concepts
+      | c   |
+      | COL |
+
+
+  Scenario: when linking an attribute that doesn't exist yet to a relation, the attribute gets created
+    Given graql define
+      """
+      define
+      residence sub relation,
+        relates resident,
+        relates place-of-residence,
+        has tenure-days,
+        key ref;
+      person plays resident;
+      address sub attribute, value string, plays place-of-residence;
+      tenure-days sub attribute, value long;
+      """
+    Given the integrity is validated
+    When graql insert
+      """
+      insert
+      $p isa person, has name "Homer", has ref 0;
+      $addr "742 Evergreen Terrace" isa address;
+      $r (resident: $p, place-of-residence: $addr) isa residence, has ref 0;
+      """
+    When the integrity is validated
+    When get answers of graql query
+      """
+      match $td isa tenure-days; get;
+      """
+    Then answer size is: 0
+    Then graql insert
+      """
+      match
+        $r isa residence;
+      insert
+        $r has tenure-days 365;
+      """
+    Then get answers of graql query
+      """
+      match $r isa residence, has tenure-days $a; get $a;
+      """
+    Then concept identifiers are
+      |     | check | value           |
+      | RES | key   | ref:0           |
+      | TEN | value | tenure-days:365 |
+    Then uniquely identify answer concepts
+      | a   |
+      | TEN |
+
+
+  Scenario: an attribute ownership currently inferred by a rule can be explicitly inserted
```

Comment:
should this be under `attribute insertion`?

Reply:
Could be either really. I think the key thing is that we're inserting an ownership, not just an attribute.